pkg/auth/dependency/auth/session_middleware.go

```diff
@@ -45,25 +49,48 @@ func (m *Middleware) resolve(rw http.ResponseWriter, r *http.Request) (AuthSessi
 	}
 	defer m.TxContext.RollbackTx()
 
-	sessionIDP, err := m.IDPSessionResolver.Resolve(rw, r)
+	session, err := m.resolveSession(rw, r)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	if sessionIDP == nil {
+	// No session credentials provided, return no error and no resolved session
+	if session == nil {
 		return nil, nil, nil
 	}
 
 	user := &authinfo.AuthInfo{}
-	if err = m.AuthInfoStore.GetAuth(sessionIDP.AuthnAttrs().UserID, user); err != nil {
+	if err = m.AuthInfoStore.GetAuth(session.AuthnAttrs().UserID, user); err != nil {
 		return nil, nil, err
 	}
 
-	event := sessionIDP.GetAccessInfo().LastAccess
-	err = m.AccessEvents.RecordAccess(sessionIDP, event)
+	event := session.GetAccessInfo().LastAccess
+	err = m.AccessEvents.RecordAccess(session, event)
 	if err != nil {
 		return nil, nil, err
 	}
 
-	return sessionIDP, user, nil
+	return session, user, nil
+}
+
+func (m *Middleware) resolveSession(rw http.ResponseWriter, r *http.Request) (AuthSession, error) {
+	isInvalid := false
+
+	// IDP session in cookie takes priority over access token in header
+	for _, resolver := range []SessionResolver{m.IDPSessionResolver, m.AccessTokenSessionResolver} {
+		session, err := resolver.Resolve(rw, r)
+		if errors.Is(err, ErrInvalidSession) {
+			// Continue to attempt resolving session, even if one of the resolver reported invalid.
+			isInvalid = true
```

Comment:
We never set it to `false` again. It is always true once it becomes true. So why don't we `return nil, ErrInvalidSession` here?